src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistryTest.java

```diff
@@ -81,9 +82,29 @@ public void codecToCQL(DataType cqlType, Object fromValue, Object expectedCqlVal
   }
 
   private static Stream<Arguments> codecToCQLTestCases() {
+    // Arguments: (CQL-type, from-caller, bound-by-driver-for-cql
+    // Note: all Numeric types accept 3 Java types: Long, BigInteger, BigDecimal
     return Stream.of(
-        Arguments.of(DataTypes.INT, BigDecimal.valueOf(100), 100) // second 100 is an int
-        );
+        // Integer types:
+        Arguments.of(DataTypes.BIGINT, -456L, -456L),
+        Arguments.of(DataTypes.BIGINT, BigInteger.valueOf(123), 123L),
+        Arguments.of(DataTypes.BIGINT, BigDecimal.valueOf(999.0), 999L),
+        Arguments.of(DataTypes.INT, -42000L, -42000),
+        Arguments.of(DataTypes.INT, BigInteger.valueOf(19000), 19000),
+        Arguments.of(DataTypes.INT, BigDecimal.valueOf(23456.0), 23456),
+        Arguments.of(DataTypes.SMALLINT, -3999L, (short) -3999),
+        Arguments.of(DataTypes.SMALLINT, BigInteger.valueOf(1234), (short) 1234),
+        Arguments.of(DataTypes.SMALLINT, BigDecimal.valueOf(10911.0), (short) 10911),
+        Arguments.of(DataTypes.TINYINT, -39L, (byte) -39),
+        Arguments.of(DataTypes.TINYINT, BigInteger.valueOf(123), (byte) 123),
+        Arguments.of(DataTypes.TINYINT, BigDecimal.valueOf(109.0), (byte) 109),
+        Arguments.of(DataTypes.VARINT, -39999L, BigInteger.valueOf(-39999)),
+        Arguments.of(DataTypes.VARINT, BigInteger.valueOf(1), BigInteger.valueOf(1)),
+        Arguments.of(
+            DataTypes.VARINT, BigDecimal.valueOf(123456789.0), BigInteger.valueOf(123456789)),
```

Comment:
I've not checked that the list here and below is exhaustive, let me know if you want it checked.

Reply:
Yeah it's not exhaustive in many ways (i.e. does not try different ranges, values), but does try to cover permutations of 3 input "plain" Java types (BigDecimal, BigInteger, Long) x CQL integer types.

So if you can sanity check, that'd be great.